app/knights/knight.py

```diff
@@ -0,0 +1,34 @@
+from .equipment import Armour, Weapon, Potion
+
+
+class Knight:
+    def __init__(
+            self,
+            name: str,
+            power: int,
+            hp: int,
+            armour: list,
+            weapon: dict,
+            potion: dict
+    ) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.armour = [Armour(**a) for a in armour]
```

Comment:
Why did you use the short name "a" in the case of the armor? It is better to use a clearer name. 
It's the same in others.